generator/sbpg/targets/resources/c/new/sbp_messages_template.h

```diff
@@ -41,114 +42,405 @@
 
 struct sbp_state;
 
-((*- for m in msgs *))
-((*- if m.fields *))
-((*- for f in m.fields *))
-((*- if f.options.fields *))
-#ifndef LIBSBP_LEGACY_(((pkg_name|upper)))_MESSAGES_H
-(((f|create_bitfield_macros(m.name))))
-#endif
-((*- endif *))
-((*- endfor *))
-((*- endif *))
+
+((* for m in msgs *))
+((*- set msg_type = m.name|convert_unpacked *))
+/*
+ *
+ *
+ *
+ ******************************************************************************
+ *
+ * SBP_(((m.name|upper)))
+ *
+ *****************************************************************************/
+
 ((*- if m.desc *))
 /** (((m.short_desc)))
  *
-((m.desc|commentify)))
+(((m.desc|commentify)))
  */
 ((*- endif *))
-((*- if m.is_real_message *))
-#ifndef LIBSBP_LEGACY_(((pkg_name|upper)))_MESSAGES_H
-#define SBP_(((m.name.ljust(max_msgid_len)))) ((('0x%04X'|format(m.sbp_id))))
-#endif
-((*- endif *))
 typedef struct {
   ((*- if m.fields *))
   ((*- for f in m.fields *))
+
+  ((* if f.desc *))
+  /**
+(((f.desc|commentify))) ((* if f.units *))[(((f.units)))]((* endif*))
+   */
+  ((*- endif *))
   ((*- if f.packing == "packed-string" *))
   sbp_(((f.options.encoding.value)))_string_t (((f.name)));
   ((*- elif f.packing == "single" *))
   (((f.basetype|convert_unpacked))) (((f.name)));
   ((*- else *))
   (((f.basetype|convert_unpacked))) (((f.name)))[(((f.max_items)))];
   ((*- if f.generate_size_fn *))
+  /**
+   * Number of elements in (((f.name)))
+   *
+   * When sending a message fill in this field with the number elements set in (((f.name))) before calling an appropriate libsbp send function
+   *
+   * When receiving a message query this field for the number of elements in (((f.name))). The value of any elements beyond the index specified in this field is undefined
+   */
   u8 (((f.size_fn)));
   ((*- endif *))
   ((*- endif *))
   ((*- endfor *))
   ((*- else *))
+  /**
+   * Do not use this field, it exists only to prevent illegal C syntax. Any value written to this field will be ignored
+   */
   char DO_NOT_USE_dummy_field_to_prevent_empty_struct;
   ((*- endif *))
-} (((m.name|convert_unpacked)));
+} (((msg_type)));
 
-((*- for f in m.fields *))
+((* for f in m.fields *))
   ((*- if f.packing == "packed-string" *))
-  void (((m.name|convert_unpacked)))_(((f.name)))_init(sbp_(((f.encoding)))_string_t *s);
-  bool (((m.name|convert_unpacked)))_(((f.name)))_valid(const sbp_(((f.encoding)))_string_t *s);
-  int (((m.name|convert_unpacked)))_(((f.name)))_strcmp(const sbp_(((f.encoding)))_string_t *a, const sbp_(((f.encoding)))_string_t *b);
-  uint8_t (((m.name|convert_unpacked)))_(((f.name)))_packed_len(const sbp_(((f.encoding)))_string_t *s);
-  uint8_t (((m.name|convert_unpacked)))_(((f.name)))_space_remaining(const sbp_(((f.encoding)))_string_t *s);
+  ((*- set prefix = msg_type + "_" + f.name *))
```

Comment:
Already mentioned on https://swift-nav.atlassian.net/wiki/spaces/~61688916/pages/1837891645/Name+location+changes+needed
Since this is an external symbol we'll do all of them in one go at the end